src/types/UiaTextRangeBase.cpp

```diff
@@ -310,16 +316,275 @@ IFACEMETHODIMP UiaTextRangeBase::ExpandToEnclosingUnit(_In_ TextUnit unit) noexc
+
+        // The text is either italic or it isn't.
+        const auto queryIsItalic{ val.boolVal };
+        if (queryIsItalic)
+        {
+            return [](const TextAttribute& attr) noexcept {
+                return attr.IsItalic();
+            };
+        }
+        else
+        {
+            return [](const TextAttribute& attr) noexcept {
+                return !attr.IsItalic();
+            };
+        }
+    }
+    case UIA_StrikethroughStyleAttributeId:
+    {
+        // Expected type: VT_I4
+        THROW_HR_IF(E_INVALIDARG, val.vt != VT_I4);
+
+        // The strikethrough style is stored as a TextDecorationLineStyle.
+        // However, The text buffer doesn't have different styles for being crossed out.
+        // Instead, we just store whether or not the text is crossed out.
+        switch (val.lVal)
+        {
+        case TextDecorationLineStyle_None:
+            return [](const TextAttribute& attr) noexcept {
+                return !attr.IsCrossedOut();
+            };
+        case TextDecorationLineStyle_Single:
+            return [](const TextAttribute& attr) noexcept {
+                return attr.IsCrossedOut();
+            };
+        default:
+            return nullptr;
+        }
+    }
+    case UIA_UnderlineStyleAttributeId:
+    {
+        // Expected type: VT_I4
+        THROW_HR_IF(E_INVALIDARG, val.vt != VT_I4);
+
+        // The underline style is stored as a TextDecorationLineStyle.
+        // However, The text buffer doesn't have that many different styles for being underlined.
+        // Instead, we only have single and double underlined.
+        switch (val.lVal)
+        {
+        case TextDecorationLineStyle_None:
+            return [](const TextAttribute& attr) noexcept {
+                return !attr.IsUnderlined() && !attr.IsDoublyUnderlined();
+            };
+        case TextDecorationLineStyle_Double:
+            return [](const TextAttribute& attr) noexcept {
+                return attr.IsDoublyUnderlined();
+            };
+        case TextDecorationLineStyle_Single:
+            return [](const TextAttribute& attr) noexcept {
+                return attr.IsUnderlined();
+            };
+        default:
+            return nullptr;
+        }
+    }
+    default:
+        return nullptr;
+    }
 }
 
+IFACEMETHODIMP UiaTextRangeBase::FindAttribute(_In_ TEXTATTRIBUTEID attributeId,
+                                               _In_ VARIANT val,
+                                               _In_ BOOL searchBackwards,
+                                               _Outptr_result_maybenull_ ITextRangeProvider** ppRetVal) noexcept
+try
+{
+    RETURN_HR_IF(E_INVALIDARG, ppRetVal == nullptr);
+    *ppRetVal = nullptr;
+
+    // AttributeIDs that require special handling
+    switch (attributeId)
+    {
+    case UIA_FontNameAttributeId:
+    {
+        RETURN_HR_IF(E_INVALIDARG, val.vt != VT_BSTR);
+        const std::wstring queryFontName{ val.bstrVal };

```

Comment:
Technically... I believe that this will truncate early if there is an embedded null in the `BSTR` as you're not using it as a structure and checking its length to populate the `std::wstring`. I think that's probably OK in this circumstance, but you may want to leave a note.